### PR TITLE
Add WSO2 APK implementation to docs

### PR DESCRIPTION
**Pull request by AmaliMatharaarachchi**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
The PR adds WSO2 APK to the list of implementations in the docs.

**Which issue(s) this PR fixes**:
None

**Does this PR introduce a user-facing change?**:
None

### CODE REVIEW

**Comment by AmaliMatharaarachchi** [comment]
@bowei @keithmattix Can you please take a look? Thanks in advance.

**Comment by arkodg** [comment]
looks good, thanks !
/approve

File: site-src/implementations.md [hunk 1]
```diff
@@ -305,6 +307,18 @@ Traefik is currently working on implementing UDP, and ReferenceGrant. Status upd
...
+### WSO2 APK
+
+[WSO2 APK][wso2-apk] is a purpose-built API management solution tailored for Kubernetes environments, delivering seamless integration, flexibility, and scalability to organizations in managing their APIs.
+
+WSO2 APK implements the Gateway API, encompassing Gateway and HTTPRoute functionalities. Additionally, it provides support for rate limiting, authentication/authorization, and analytics/observability through the use of Custom Resources (CRs).
+
+For up-to-date information on the supported version and features of the Gateway API, please refer to the [APK Gateway documentation][apk-doc]. If you have any questions or would like to contribute, feel free to create [issues or pull requests][repo]. Join our [Discord channel][discord] to connect with us and engage in discussions.
+
+[wso2-apk]:https://apk.docs.wso2.com/en/latest/
+[apk-doc]:https://apk.docs.wso2.com/en/latest/create-api/create-api-overview/
```
[the remark this answers is not in the document]
Comment (AmaliMatharaarachchi) [reply site-src/implementations.md]: @robscott Thanks for the suggestion. I have updated the PR with a more suitable link.

**Comment by robscott** [comment]
Thanks @AmaliMatharaarachchi!

/lgtm
/approve

**Pull request merged**